components/onboarding/onboardingContext.js

```diff
@@ -0,0 +1,204 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { driver } from 'driver.js'; // import driver.js
+
+import 'driver.js/dist/driver.css'; //import css
+import onboardingScriptGeneric from './onboardingScriptGeneric.json';
+import onboardingScriptCustom from './onboardingScriptCustom.json';
+
+const OnboardingContext = createContext(undefined);
+
+function useOnboardingProvider() {
+    const router = useRouter();
+    const [active, setActive] = useState(false);
+
+    const [currentRoute, setCurrentRoute] = useState('');
+    const [nextRouteName, setNextRouteName] = useState('');
+
+    const [prevRouteName, setPrevRouteName] = useState('');
+
+    const [currentRouteIndex, setCurrentRouteIndex] = useState(0);
+
+    const [currentSteps, setCurrentSteps] = useState([]);
+
+    const [currentStep, setCurrentStep] = useState(0);
```

Comment:
are we using `currentStep` anywhere? If so, I think we should rename `currentSteps` (plural) to `currentRouteSteps` or something similar, it's hard to distinguish them both when reading the code.
But i don't think `currentStep` is being used.